src/main/java/net/masterthought/cucumber/CucumberReportGeneratorMojo.java

```diff
@@ -111,7 +111,7 @@ public void execute() throws MojoExecutionException {
             getLog().info("About to generate Cucumber report.");
             Reportable report = reportBuilder.generateReports();
 
-            if (checkBuildResult && report == null) {
+            if (checkBuildResult && (report == null || report.getFailedSteps() > 0)) {
```

Comment:
I don't want to fail the build if some tests failed. The responsibility of this tool is to generate report and don't change flow unless report could not be generated

Reply:
Would you accept the request if there is some flag to activate this function? It's really important to me that the report is created even if the build breaks, and also that the build break when it should break.

Reply:
Would it be enough for you? https://github.com/jenkinsci/cucumber-reports-plugin/blob/master/src/main/java/net/masterthought/jenkins/CucumberReportPublisher.java#L321

Reply:
Can we release this please?

Reply:
Will be with next release

Reply:
When is the next release?

Reply:
Hi, i am facing this issue in android + cucumber + Espresso combination.
Can anyone please suggest a solution? How can i modify the gradle file so that build is not failed when a test is failed.

Reply:
There is readme about maven but I'm not familiar much with Gradle

Reply:
thanks for your respone @damianszczepanik. i was able to get rid of the issue by adding the below lines in build.gradle.
project.gradle.taskGraph.whenReady {
    connectedDebugAndroidTest {
        ignoreFailures = true
    }
}